profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -0,0 +1,167 @@
+import com.splunk.opentelemetry.profiler.ProfilingDataType;
+import com.splunk.opentelemetry.profiler.allocation.sampler.AllocationEventSampler;
+import com.splunk.opentelemetry.profiler.exporter.PprofLogDataExporter;
+import com.splunk.opentelemetry.profiler.pprof.Pprof;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.resources.Resource;
+import java.time.Instant;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedFrame;
+import jdk.jfr.consumer.RecordedMethod;
+import jdk.jfr.consumer.RecordedStackTrace;
+import jdk.jfr.consumer.RecordedThread;
+
+public class PprofAllocationEventExporter implements AllocationEventExporter {
+  private final LogProcessor logProcessor;
+  private final Resource resource;
+  private final DataFormat allocationDataFormat;
+  private final PprofLogDataExporter pprofLogDataExporter;
+  private Pprof pprof = createPprof();
+
+  private PprofAllocationEventExporter(Builder builder) {
+    this.logProcessor = builder.logProcessor;
+    this.resource = builder.resource;
+    this.allocationDataFormat = builder.allocationDataFormat;
+    this.pprofLogDataExporter =
+        new PprofLogDataExporter(
+            builder.logProcessor,
+            builder.resource,
+            ProfilingDataType.ALLOCATION,
+            builder.allocationDataFormat);
+  }
+
+  @Override
+  public void export(RecordedEvent event, AllocationEventSampler sampler, SpanContext spanContext) {
+    RecordedStackTrace stackTrace = event.getStackTrace();
+    if (stackTrace == null) {
+      return;
+    }
+
+    long allocationSize = event.getLong("allocationSize");
+
+    Sample.Builder sample = Sample.newBuilder();
+    sample.addValue(allocationSize);
+    // XXX StackSerializer limits stack depth. Although I believe jfr also limits it should we
+    // attempt to limit the depth here too?
+    for (RecordedFrame frame : stackTrace.getFrames()) {
+      RecordedMethod method = frame.getMethod();
+      if (method == null) {
+        sample.addLocationId(pprof.getLocationId("unknown", "unknown.unknown", -1));
+      } else {
+        sample.addLocationId(
+            pprof.getLocationId(
+                "unknown", // file name is not known
+                method.getType().getName() + "." + method.getName(),
+                frame.getLineNumber()));
+      }
+    }
+
+    String eventName = event.getEventType().getName();
+    pprof.addLabel(sample, SOURCE_EVENT_NAME, eventName);
+    Instant time = event.getStartTime();
+    pprof.addLabel(sample, SOURCE_EVENT_TIME, time.toEpochMilli());
+
+    RecordedThread thread = event.getThread();
+    if (thread.getJavaThreadId() != -1) {
+      pprof.addLabel(sample, THREAD_ID, thread.getJavaThreadId());
+      pprof.addLabel(sample, THREAD_NAME, thread.getJavaName());
+    }
+    pprof.addLabel(sample, THREAD_NATIVE_ID, thread.getOSThreadId());
```

Comment:
What about explicitly adding `pprof.addLabel(sample, THREAD_STATUS, "RUNNABLE");` here, so that we don't have to add special handling on the backend?